File: Controllers/MembershipController.cpp
#include "MembershipController.h"
#include <iostream>
using namespace std;

MembershipController::MembershipController(Database& db) : db(db) {}

void MembershipController::getAll() {
    cout << "\n--- ALL MEMBERSHIP PLANS ---\n";
    db.loadPlans();
}

void MembershipController::getOne(int planID) {
    cout << "\n--- PLAN DETAILS ---\n";
    db.getPlanByID(planID);
}

void MembershipController::create(string planName, double price, int durationMonths) {
    Membership m(planName, durationMonths, price);
    db.savePlan(m);
    cout << "✔ Plan '" << planName << "' created successfully.\n";
}

void MembershipController::update(int planID, string planName, double price, int durationMonths) {
    db.updatePlan(planID, planName, price, durationMonths);
    cout << "✔ Plan ID " << planID << " updated successfully.\n";
}

void MembershipController::remove(int planID) {
    db.deletePlan(planID);
    cout << "✔ Plan ID " << planID << " deleted.\n";
}

File: Controllers/MembershipController.js
// MembershipController.js
// Replace `plans` array with backend API calls later

let plans = [];

const MembershipController = {

    getAll() {
        return plans;
    },

    getOne(planID) {
        return plans.find(p => p.planID === planID) || null;
    },

    create(planName, price, durationMonths) {
        const planID = plans.length > 0 ? plans[plans.length - 1].planID + 1 : 1;
        plans.push({ planID, planName, price, durationMonths });
        return true;
    },

    update(planID, planName, price, durationMonths) {
        const index = plans.findIndex(p => p.planID === planID);
        if (index === -1) {
            alert("Plan not found!");
            return false;
        }
        plans[index] = { planID, planName, price, durationMonths };
        return true;
    },

    remove(planID) {
        const index = plans.findIndex(p => p.planID === planID);
        if (index === -1) {
            alert("Plan not found!");
            return false;
        }
        plans.splice(index, 1);
        return true;
    }
};